webapp/src/ts/services/transitions/user-replace.transition.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@angular/core';
+
+import { DbService } from '@mm-services/db.service';
+import { Transition } from '@mm-services/transitions/transition';
+import { UserReplaceService } from '@mm-services/user-replace.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserReplaceTransition extends Transition {
+  constructor(
+    private dbService: DbService,
+    private userReplaceService: UserReplaceService,
+  ) {
+    super();
+  }
+
+  readonly name = 'user_replace';
+
+  init() {
+    // TODO Update this to load configured forms based on passed in settings
+    return true;
+  }
+
+  /**
+   * @param {Array<Doc>} docs - docs to be saved
+   * @return {Boolean} - whether any of the docs from the batch should be processed
+   */
+  filter(docs) {
+    return docs.filter(doc => doc.type === 'data_record').length;
+  }
```

Comment:
The JSDoc is neat! Thankfully we're using typescript for the webapp, let's use it :grinning: 

```suggestion
  filter(docs: Doc[]) {
    return docs.filter(doc => doc.type === 'data_record').length > 0;
  }
```

Reply:
Good call!  I have tweaked some things to hopefully make this better TS.   (Though as you can tell, I am no TypeScript expert yet, so please let me know if you thing there are other ways we can make this better...)